coordinator/schedulepod/main.go

```diff
@@ -0,0 +1,694 @@
+	}
+	for _, value := range testOfflineTableValues {
+		if err := featureTable.Write(value); err != nil {
+			return fmt.Errorf("could not write to offline feature table")
+		}
+	}
+	offline_label := provider.ResourceID{Name: labelName, Variant: "", Type: provider.Label}
+	labelTable, err := my_offline.CreateResourceTable(offline_label, schemaInt)
+	if err != nil {
+		return fmt.Errorf("could not create label table: %v", err)
+	}
+	for _, value := range testOfflineTableValues {
+		if err := labelTable.Write(value); err != nil {
+			return fmt.Errorf("could not write to offline label table")
+		}
+	}
+	originalTableName := createSafeUUID()
+	if err := CreateOriginalPostgresTable(originalTableName); err != nil {
+		return err
+	}
+	sourceName := createSafeUUID()
+	if err := createTrainingSetWithProvider(client, serialPGConfig, sourceName, featureName, labelName, tsName, originalTableName, "*/1 * * * *"); err != nil {
+		return fmt.Errorf("could not create training set %v", err)
+	}
+	ctx := context.Background()
+	tsID := metadata.ResourceID{Name: tsName, Variant: "", Type: metadata.TRAINING_SET_VARIANT}
+	tsCreated, err := client.GetTrainingSetVariant(ctx, metadata.NameVariant{Name: tsName, Variant: ""})
+	if err != nil {
+		return fmt.Errorf("could not get training set")
+	}
+	if tsCreated.Status() != metadata.CREATED {
+		return fmt.Errorf("Training set not set to created with no coordinator running")
+	}
+	kubeJobSpawner := coordinator.KubernetesJobSpawner{}
+	coord, err := coordinator.NewCoordinator(client, logger, cli, &kubeJobSpawner)
+	if err != nil {
+		return fmt.Errorf("Failed to set up coordinator")
+	}
+	sourceID := metadata.ResourceID{Name: sourceName, Variant: "", Type: metadata.SOURCE_VARIANT}
+	if err := coord.ExecuteJob(metadata.GetJobKey(sourceID)); err != nil {
+		return err
+	}
+	go func() {
+		if err := coord.WatchForUpdateEvents(); err != nil {
+			logger.Errorf("Error watching for new update events: %v", err)
+		}
+	}()
+	if err := coord.ExecuteJob(metadata.GetJobKey(tsID)); err != nil {
+		return err
+	}
+	time.Sleep(70 * time.Second)
+	jobClient, err := runner.NewKubernetesJobClient(runner.GetCronJobName(tsID), runner.Namespace)
+	if err != nil {
+		return err
+	}
+	cronJob, err := jobClient.GetCronJob()
+	if err != nil {
+		return err
+	}
+	lastExecutionTime := cronJob.Status.LastSuccessfulTime
+	if lastExecutionTime.IsZero() {
+		return fmt.Errorf("job did not execute in time")
+	}
+	// tsUpdated, err := client.GetTrainingSetVariant(ctx, metadata.NameVariant{Name: tsName, Variant: ""})
+	// if err != nil {
+	// 	return err
+	// }
+	// tUpdateStatus := tsUpdated.UpdateStatus()
+	// tsLastUpdated := tUpdateStatus.LastUpdated
+	// if tsLastUpdated.AsTime().IsZero() {
+	// 	return fmt.Errorf("Scheduler did not update training set")
+	// }
+	return nil
+}
+
+func testScheduleFeatureMaterialization() error {
+	if err := runner.RegisterFactory(string(runner.COPY_TO_ONLINE), runner.MaterializedChunkRunnerFactory); err != nil {
```

Comment:
Can all these functions be broken into sub functions? Theres a ton going on here

Reply:
sure